### PR TITLE
Add webhook library and updating scripts

### LINKED ISSUE #1: Add webhooks to update specific packages and branches rather than updating on cron
This will match how packagist updates packages. It will be much more efficient to update and much more responsive.

This will require a backend PHP process to parse the webhook and call the appropriate satis update commands.

### PR DESCRIPTION
**Pull request by charleswaddell**
Fixes #1

### CODE REVIEW

**Comment by charleswaddell** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;vIJ97xok&#x2F;1010-0-25d-use-web-hook-to-update-satis-package-cache"> 0.25d - Use web hook to update satis package cache</a>

File: composer.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
 	"license": "MIT",
 	"minimum-stability": "dev",
 	"require": {
-		"composer/satis": "*"
+		"composer/satis": "*",
+		"silverorange/github-webhook-handler": "^0.0.1"
```
Comment (gauthierm) [inline composer.json]: Caret operator should not be used for pre 1.x versions.

File: www/webhook/index.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+require_once __DIR__.'/../../vendor/autoload.php';
```
Comment (gauthierm) [inline www/webhook/index.php]: ws around `.`

File: www/webhook/index.php [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+$app = new \silverorange\GitHubWebhook\Handler();
+$app(__DIR__.'/../../webhook.json');
```
Comment (gauthierm) [inline www/webhook/index.php]: ws around `.`

File: composer.json [hunk 4]
```diff
@@ -6,6 +6,7 @@
 	"license": "MIT",
 	"minimum-stability": "dev",
 	"require": {
-		"composer/satis": "*"
+		"composer/satis": "*",
+		"silverorange/github-webhook-handler": "0.0.1"
```
Comment (gauthierm) [inline composer.json]: Should be `>=0.0.1` for releases less than 1.0

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,6 @@
-www/
+www/dist
+www/include
+www/index.html
+www/packages.json
```
Comment (gauthierm) [inline .gitignore]: why do we ignore all these files?

File: bin/update-repo.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+path=`dirname $0`
+bin="$path/../vendor/bin/satis"
+json="$path/../silverorange.json"
+www="$path/../www"
+repo=$1
+
+php -d date.timezone=UTC -d memory_limit=-1 $bin build --repository-url $repo $json $www
```
Comment (gauthierm) [inline bin/update-repo.sh]: can we include a branch/tag name here to make it even faster?

File: .gitignore (hunk 5 above)
Comment (charleswaddell) [inline .gitignore]: We were ignoring them before. They are generated but Satis. We don't want to ignore the webhook file that now sites in `www` though.

File: bin/update-repo.sh (hunk 6 above)
Comment (charleswaddell) [inline bin/update-repo.sh]: Doesn't [look like it](https://getcomposer.org/doc/articles/handling-private-packages-with-satis.md#partial-updates).

Comment (gauthierm) [inline bin/update-repo.sh]: @charleswaddell and I discussed specifying the package name to update here and decided for a first pass to omit the package name.

**Pull request merged**